lib/Normalize_extended_ast.ml

```diff
@@ -159,18 +159,42 @@ let docstring conf =
   let normalize_code = normalize_code conf mapper in
   docstring conf ~normalize_code
 
+let zip_align (l1, l2) =
+  let rec aux acc l1 l2 =
+    match (l1, l2) with
+    | [], _ -> List.rev_map_append ~f:(fun x -> `Added x) l2 acc
+    | _, [] -> List.rev_map_append ~f:(fun x -> `Dropped x) l1 acc
+    | h1 :: t1, h2 :: t2 -> (
+      match Migrate_ast.Location.compare (Cmt.loc h1) (Cmt.loc h2) with
+      | 0 -> aux (`Modified (h1, h2) :: acc) t1 t2
+      | x when x < 0 -> aux (`Dropped h1 :: acc) t1 l2
+      | _ -> aux (`Added h2 :: acc) l1 t2 )
+  in
+  List.rev (aux [] l1 l2)
+
+let diff ~f x y =
+  let loc = function Either.First x | Either.Second x -> x.Cmt.loc in
+  Set.symmetric_diff (f x) (f y)
+  |> Sequence.to_list
+  |> List.sort ~compare:(fun x y ->
+         Migrate_ast.Location.compare (loc x) (loc y) )
+  |> List.partition_map ~f:Fn.id
+  |> zip_align
```

Comment:
It would be good to have a comment to explain this operation (partition then align)

Reply:
done

Reply:
That's useful but it would be nice to write why the two lists need to be aligned (and why `symmetric_diff` is not enough).

Reply:
I realized a lot of the code I added in this function was not needed, I added some comments for clarity.

So [`Modified] errors on regular comments are not being used, I'm leaving the type as is for now, to stay generic.